main.go

```diff
@@ -251,3 +326,19 @@ func rawSetuid(uid int) error {
 
 	return nil
 }
+
+func (handler *BuildHandler) saveNewBuild(repoName string) BuildInfo {
+	handler.buildListMutex.Lock()
+	defer handler.buildListMutex.Unlock()
+	info := BuildInfo{
+		repository: repoName,
+		startedAt:  time.Now(),
+		status:     "in progress",
+
+		pointer: handler.lastBuild,
+	}
+	handler.lastBuild.Value = info
+
+	handler.lastBuild = handler.lastBuild.Prev()
```

Comment:
Moving back for reversed order when listing. As `Ring.Do(...)` go in forward direction, we need to add it backward for LIFO order in list.